### PR TITLE
feat(simulate): improve scoring and sorting

### PR DESCRIPTION
**Pull request by Xiphe**
<a href="https://teamsgame.agilepainrelief.com/preview/simulate-better-sorting-custom-scoring">published to gh-pages</a><hr />

Improves on scoring and sorting for simulations and hopefully makes them easier to understand and customize

### CODE REVIEW

File: src/simulate/ResultsApp.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useMemo, useState, ReactNode } from 'react';
```
Comment (Xiphe) [inline src/simulate/ResultsApp.tsx:1]: Refactored to reuse scores from config

File: src/simulate/index.ts [hunk 2]
```diff
@@ -138,98 +127,99 @@ for (let index = 0; index < simulationsToRun; index++) {
...
-      selectedActions: state.pastRounds.reduce(
-        (i, { selectedGameActionIds }) => i + selectedGameActionIds.length,
-        0,
-      ),
-      ocurredGremlins: state.pastRounds.filter(
-        ({ gremlin }) => gremlin !== null,
-      ).length,
-      storiesAttempted: storiesAttempted,
-      storiesCompleted: storiesCompleted,
-    },
```
Comment (Xiphe) [inline src/simulate/index.ts:165]: These are now calculated via scorings.tsx

File: src/simulate/index.ts [hunk 3]
```diff
@@ -138,98 +127,99 @@ for (let index = 0; index < simulationsToRun; index++) {
...
+      }
+
+      /* If all scores are same, fall back to id */
+      return a - b;
+    });
+    if (bestGames[key].length > (createCSVFor.includes(key) ? STORE_BEST : 1)) {
+      /* cut of worst of the best games */
+      const removed = bestGames[key].splice(0, 1);
+      irrelevant.push(...removed);
+    }
```
Comment (Xiphe) [inline src/simulate/index.ts:188]: Hope this is easier to understand now

File: src/simulate/index.ts [hunk 4]
```diff
@@ -138,98 +127,99 @@ for (let index = 0; index < simulationsToRun; index++) {
...
+      /* If all scores are same, fall back to id */
+      return b - a;
+    });
+    if (
+      worstGames[key].length > (createCSVFor.includes(key) ? STORE_WORST : 1)
+    ) {
+      /* cut of worst of the best games */
+      const removed = worstGames[key].splice(0, 1);
+      irrelevant.push(...removed);
+    }
```
Comment (Xiphe) [inline src/simulate/index.ts:216]: Same as above, just the other way round

File: src/simulate/simulateConfig.ts [hunk 5]
```diff
@@ -9,12 +19,54 @@ export const simulateConfig: OverwritableConfig = {
...
+  storiesAttempted,
+  storiesCompleted,
+};
+
+type ScoringKey = keyof typeof scorings;
+type SortDir = 'asc' | 'desc';
+type Sort = Partial<Record<ScoringKey, SortDir>>;
+
+/* only create csv output for these scores */
+export const createCSVFor: ScoringKey[] = ['combinedScore'];
```
Comment (Xiphe) [inline src/simulate/simulateConfig.ts:55]: Will only output/update `combinedScore.csv` feel free to add the others back however you like

**Review by mlevison — APPROVED** [review]

**Pull request merged**